gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CredentialServiceImpl.java

```diff
@@ -140,7 +148,22 @@ public Completable delete(String id) {
         LOGGER.debug("Delete credential {}", id);
         return credentialRepository.findById(id)
                 .switchIfEmpty(Maybe.error(new CredentialNotFoundException(id)))
-                .flatMapCompletable(email -> credentialRepository.delete(id))
+                .flatMapCompletable(credential ->
+                        userService.findById(credential.getUserId()).flatMapCompletable(user -> {
+                            Optional<EnrolledFactor> fido2Factor = user.getFactors()
```

Comment:
User with Credentials may not have factors, in this case the `getFactors` will return null and we have to manage the case. 
Currently we are failing with 500 Internal Error due to NullPointerException